### ISSUE TITLE
HW 4 Topic

### ISSUE THREAD

**Issue by vhall415** [mikeizbicki/ucr-cs100#940]
I'd like to write a gdb tutorial/cheatsheet on how to get the most out of debugging with gdb.

**Comment by mikeizbicki** [comment]
There is already both a gdb tutorial and a cheatsheet.  The tutorial especially has some room or improvement, but how, specifically, would you be improving it?

**Comment by vhall415** [comment]
You're right. I noticed the file in the textbook section containing all the stuff I wanted to expand on with gdb that wasn't there in the lab. Instead, I'd like to write my tutorial on template meta programming. I find using templates very useful since the template can work for any type, yet I've noticed some people struggle with the template concept. Plus, I've done some research into using templates and have found different ways of using them that I didn't know of before such as how to use them in structs and how those structs can be used recursively to meet a desired end.
